opencti-platform/opencti-graphql/src/domain/connector.js

```diff
@@ -130,6 +132,41 @@ export const connectorDelete = async (context, user, connectorId) => {
   await notify(BUS_TOPICS[ABSTRACT_INTERNAL_OBJECT].DELETE_TOPIC, element, user);
   return element.internal_id;
 };
+
+export const connectorTriggerUpdate = async (context, user, connectorId, input) => {
+  const conn = await storeLoadById(context, user, connectorId, ENTITY_TYPE_CONNECTOR);
+  if (!conn) {
+    throw FunctionalError('Cant find element to update', { id: connectorId, type: ENTITY_TYPE_CONNECTOR });
+  }
+  if (!['INTERNAL_ENRICHMENT', 'INTERNAL_IMPORT_FILE'].includes(conn.connector_type)) {
+    throw FunctionalError('Update is only possible on internal enrichment or import file connectors types');
+  }
+  const supportedInputKeys = ['connector_trigger_filters', 'auto'];
+  if (input.some((item) => !supportedInputKeys.includes(item.key))) {
+    throw FunctionalError(`Update is only possible on these input keys: ${supportedInputKeys.join(', ')}`);
+  }
+  const filtersItem = input.find((item) => item.key === 'connector_trigger_filters');
+  if (filtersItem?.value) {
+    const jsonFilters = JSON.parse(filtersItem.value);
+    if (isFilterGroupNotEmpty(jsonFilters)) {
+      // our stix matching is currently limited, we need to validate the input filters
+      validateFilterGroupForStixMatch(jsonFilters);
+    } else {
+      filtersItem.value = ''; // empty filter
+    }
+  }
+  const { element } = await updateAttribute(context, user, connectorId, ENTITY_TYPE_CONNECTOR, input);
```

Comment:
I'm not sure how the cache currently works, but shouldn't resetCacheForEntity be called for connector entity type? During my tests, it looks like the filter isn't taken into account without having to restart the backend

Reply:
the cache works by listening to update events, which are sent here `notify(BUS_TOPICS[ENTITY_TYPE_CONNECTOR].EDIT_TOPIC, element, user);`